storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -104,7 +104,7 @@ public final class RemoteLogManagerConfig {
         (name, value) -> { 
             if ((int) value < -1 || (int) value == 0) throw new ConfigException(name, value, "Value can be -1 or greater than 0"); 
         },
-        () -> REMOTE_LOG_MANAGER_THREAD_POOL_FALLBACK
+        () -> "[-1,1,...]"
```

Comment:
From the jdk javadoc, It looks like the [ScheduledThreadPoolExecutor](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ScheduledThreadPoolExecutor.html#%3Cinit%3E(int)) can allow 0 as the corePoolSize. Maybe we can directly put the validator as `atLeast(-1)`? WDYT?

Reply:
I agree on this. @satishd wdyt?

Reply:
What is the usecase for core pool size as 0 that is beneficial for users for these configurations? 
if somebody sets it as 0, it will be inefficient as it needs to create a thread every time.

Reply:
> What is the usecase for core pool size as 0 that is beneficial for users for these configurations? if somebody sets it as 0, it will be inefficient as it needs to create a thread every time.

I think it's the use's responsibility to set the config. But I'm fine we disallow the value 0 here if we don't think it makes sense.

Reply:
Thinking again about this. It hardly makes sense to set it to zero in this context.